### PR TITLE
Add authentication to OpenAPI schema.

### PR DESCRIPTION
**Pull request by godwinpang**
This allows for client codegen to have authorization features.

### CODE REVIEW

File: src/controllers/EventController.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ export class EventController {
 
   @Post('/')
   @ResponseSchema(EventResponse)
+  @OpenAPI({ security: [{ TokenAuth: [] }] })
```
Comment (thaigillespie) [inline src/controllers/EventController.ts:45]: What is TokenAuth?

File: src/routers/DocsRouter.ts [hunk 2]
```diff
@@ -29,6 +32,12 @@ const openAPISpec = routingControllersToSpec(rcMetadataStorage, rcOptions, {
   },
 });
 
+openAPISpec.components.securitySchemes = {};
+openAPISpec.components.securitySchemes[SecuritySchemeName] = {
+  type: 'http',
+  scheme: 'bearer',
```
Comment (thaigillespie) [inline src/routers/DocsRouter.ts:38]: What is 'bearer'?

File: src/controllers/EventController.ts (hunk 1 above)
Comment (godwinpang) [reply src/controllers/EventController.ts:45]: it's the security scheme I defined in docs router; time for docs :p

File: src/routers/DocsRouter.ts (hunk 2 above)
Comment (godwinpang) [reply src/routers/DocsRouter.ts:38]: https://swagger.io/docs/specification/authentication/bearer-authentication/
will add comment

**Pull request merged**